clients/drcachesim/analyzer.cpp

```diff
@@ -339,6 +339,16 @@ analyzer_tmpl_t<RecordType, ReaderType>::init_scheduler_common(
             uint64_t filetype = scheduler_.get_stream(i)->get_filetype();
             VPRINT(this, 2, "Worker %d filetype %" PRIx64 "\n", i, filetype);
             if (TESTANY(OFFLINE_FILE_TYPE_CORE_SHARDED, filetype)) {
+                if (i == 0 && shard_type_ == SHARD_BY_CORE) {
+                    // This is almost certainly user error.
+                    // Better to exit than risk user confusion.
+                    // Ideally this could be detected in analyzer_multi but that is
+                    // not simple so today the user has to re-run.
+                    error_string_ =
+                        "Re-scheduling a core-sharded-on-disk trace is generally a "
+                        "mistake; re-run with -no_core_sharded.\n";
+                    return false;
```

Comment:
Also mention (here or elsewhere) the discussion that we cannot simply proceed thread-sharded if -core_sharded is set on a csod, because there may potentially be differences in scheduler options

Reply:
Done, and in #7045.